bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/DefaultModelResourceManager.java

```diff
@@ -76,6 +84,7 @@
    protected ModelWatchersManager watchersManager;
    protected final Map<URI, ResourceSet> resourceSets = Maps.newLinkedHashMap();
    protected final Map<ResourceSet, ModelServerEditingDomain> editingDomains = Maps.newLinkedHashMap();
+   private final Lock resourceSetsLock = new ReentrantLock();
```

Comment:
I wonder if we should expose that to increase extensibility? If the public methods are overriden (e.g. `closeResource` or `getResourceSet`), adopters won't be able to reuse this.

Reply:
Good point. I generally try to avoid exposing things like this directly to subclasses, but I understand that (and am still getting used to) this project's policy is more or less to make everything accessible to subclasses via `protected` visibility. At least it is not changeable, so I am not opposed.

Alternatively, one could require subclasses to use the `runResourceSetAction(Runnable)` method, which would have the same effect, but that API is meant for other components to use and is not supposed to be the kind of implementation detail that subclasses normally access. So that would not be a good solution.